llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -161,6 +165,8 @@ bool VPRecipeBase::mayHaveSideEffects() const {
     Function *Fn = cast<VPWidenCallRecipe>(this)->getCalledScalarFunction();
     return mayWriteToMemory() || !Fn->doesNotThrow() || !Fn->willReturn();
   }
+  case VPWidenIntrinsicSC:
+    return false;
```

Comment:
Same

Reply:
Yep, updated to store read/write/sideeffcts in recipe.